Platform/UTMJailbreak.c

```diff
@@ -32,33 +36,62 @@ kern_return_t catch_exception_raise(mach_port_t exception_port,
                                     exception_type_t exception,
                                     exception_data_t code,
                                     mach_msg_type_number_t code_count) {
-    NSLog(@"Caught exception %d (this should be EXC_SOFTWARE), with code 0x%x (this should be EXC_SOFT_SIGNAL) and subcode %d. Forcing suicide.", exception, *code, code[1]);
+    fprintf(stderr, "Caught exception %d (this should be EXC_SOFTWARE), with code 0x%x (this should be EXC_SOFT_SIGNAL) and subcode %d. Forcing suicide.", exception, *code, code[1]);
     // _exit doesn't seem to work, but this does. ¯\_(ツ)_/¯
     return KERN_FAILURE;
 }
 
-void *exception_handler(void *argument) {
+static void *exception_handler(void *argument) {
     mach_port_t port = *(mach_port_t *)argument;
     mach_msg_server(exc_server, 2048, port, 0);
     return NULL;
 }
 
-static BOOL am_i_being_debugged() {
+static bool am_i_being_debugged() {
     int mib[] = {CTL_KERN, KERN_PROC, KERN_PROC_PID, getpid()};
     struct kinfo_proc info = {};
     size_t size = sizeof(info);
     return !sysctl(mib, sizeof(mib) / sizeof(*mib), &info, &size, NULL, 0) && !!(info.kp_proc.p_flag & P_TRACED);
 }
 
-int main(int argc, char * argv[]) {
-    BOOL debugged = am_i_being_debugged();
+
+bool jb_has_jit_entitlement(void) {
```

Comment:
Technically you should probably read your entitlements ([how iSH does it](https://github.com/ish-app/ish/blob/ce0fd919f6ca7980930c86d24e1dc480a0276e59/app/AppGroup.m#L31)) in case Apple removes the ability to unmap your JIT region and get a new one, but I guess this works for now.

Reply:
Yeah, we'll cross that bridge when we get there. I don't want to add that much code.